test/integration/auth/node_test.go

```diff
@@ -492,19 +501,21 @@ func TestNodeAuthorizer(t *testing.T) {
 
 	expectForbidden(t, createNode2NormalPod(nodeanonClient))
 	// mirror pod and self node lifecycle is allowed
+	expectAllowed(t, createNode2(node2Client))
+	expectAllowed(t, updateNode2Status(node2Client))
+	expectForbidden(t, createNode2MirrorPod(nodeanonClient))
+	expectForbidden(t, deleteNode2MirrorPod(nodeanonClient))
+	expectForbidden(t, createNode2MirrorPod(node1Client))
+	expectNotFound(t, deleteNode2MirrorPod(node1Client))
+	// create a pod as an admin to add object references
+	expectAllowed(t, createNode2NormalPod(superuserClient))
+
 	expectAllowed(t, createNode2MirrorPod(node2Client))
```

Comment:
Was this passing before?

Reply:
Hmm, it's because when the node authorizer & noderestriction plugins verify that the caller is a node, they don't actually check that it's a real node.

@liggitt was that intentional? It seems like a good check to add.

Reply:
Ah, I suppose it's because the node client needs to be able to create the node itself. Since the node client has that permission, it probably doesn't add much security to require the node to exist, but it still seems like a reasonable thing to require, since a node should never make any requests before it exists.

Reply:
@timstclair 
Node2 is deleted on line 518 (below this line) - it exists as of this check.
Can you clarify what assertion you want added here ?